R-packages/modeltools/R/matrix.R

```diff
@@ -87,84 +82,133 @@ create_train_and_predict_matrices <- function(lagged_df, ahead, training_window_
         # Need to recompute training_window_size 
         training_window_size <- training_window_size + max(ahead) - min(ahead)
         
-        out <- list()
-        
         # Find the last possible date of training data
         # (corresponds to the smallest ahead)
-        response_end_date <- lagged_df %>%
-            select(time_value, 
-                   tidyselect::starts_with(sprintf("response+%i:", min(ahead)))) %>%
-            tidyr::drop_na() %>%
-            summarize(max(time_value)) %>%
-            pull()
-        train_end_date <- min(max(lagged_df$time_value), response_end_date)
+        train_end_date <- get_train_end_date(lagged_df, ahead)
         
         # Training matrices
-        out$train_x <- lagged_df %>%
-            select(geo_value, time_value, tidyselect::starts_with("value")) %>%
-            filter(between(time_value,
-                           train_end_date - training_window_size + 1,
-                           train_end_date)) %>%
-            select(-c(geo_value, time_value)) %>%
-            as.matrix()
-        out$train_y <- lagged_df %>%
-            filter(between(time_value,
-                           train_end_date - training_window_size + 1,
-                           train_end_date)) %>%
-            select(tidyselect::starts_with(paste0("response+", ahead, ":"))) %>%
-            as.matrix()
-        
-        # Add prediction matrices and training end date
-        out$predict_x <- predict_x
-        out$predict_geo_values <- predict_geo_values
-        out$train_end_date <- train_end_date
-        
-        return(out)
+        all_out[[1]] <- create_train_matrices(lagged_df, ahead, 
+                                              training_window_size,
+                                              train_end_date)
     }
+    
+    # Add prediction matrices / info
+    for (i in seq_along(all_out)) {
+        all_out[[i]]$predict_x <- predict_x
+        all_out[[i]]$predict_geo_values <- predict_geo_values
+        all_out[[i]]$predict_time_value <- predict_time_value
+    }
+    
+    if (length(ahead) == 1 || aheads_separate == FALSE) {
+        return(all_out[[1]])
+    } else {
+        names(all_out) <- paste0("ahead+", ahead)
+        return(all_out)
+    }
+}
+
+#' Get last possible date of training data for given aheads.
+#' 
+#' Returns the last possible date of the training data for a given set of
+#' aheads. If more than one ahead is given, the date return corresponds to the
+#' last possible date corresponding to the smallest ahead value. This is because
+#' the smallest ahead value will have the latest possible date.
+#'
+#' @param lagged_df Data frame of lagged data as in `create_train_and_predict_matrices()`.
+#' @param ahead Number of days ahead to forecast. Can be a single positive 
+#'     integer or a vector of positive integers.
+#' 
+#' @return Single date corresponding to the last possible date of the training
+#' data.
```

Comment:
I appreciate the comments, but my memory is that Roxygen2 gets converted to an .Rd file even if the function is not exported. If that's the case, better to convert `#'` to `#`.

Reply:
Yes, this does get converted to an .Rd file. Personally I prefer more documentation than less, even for internal functions that are not exported (the glmnet package does this as well). But if you prefer not to have the .Rd file I can change it. Let me know.